extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

```diff
@@ -221,13 +192,13 @@ private static void setGrantClientParams(OidcClientConfig oidcConfig, MultiMap g
 
     private static Uni<String> discoverTokenEndpoint(WebClient client, String authServerUrl) {
         String discoveryUrl = authServerUrl + "/.well-known/openid-configuration";
-        return client.get(discoveryUrl).send().onItem().transformToUni(resp -> {
+        return client.getAbs(discoveryUrl).send().onItem().transform(resp -> {
             if (resp.statusCode() == 200) {
                 JsonObject json = resp.bodyAsJsonObject();
-                return Uni.createFrom().item(json.getString("token_endpoint"));
+                return json.getString("token_endpoint");
             } else {
                 LOG.tracef("Discovery has failed, status code: %d", resp.statusCode());
-                return Uni.createFrom().nullItem();
+                throw new OidcClientException("Token endpoint discovery has failed");
```

Comment:
@sd-f This null is taken care of a bit later [here](https://github.com/quarkusio/quarkus/blob/351778d3ce0aa8a44cc45328ebdc44c293d2f1bb/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java#L119).
We can throw the exception now but then we'd have to refactor the uni code catching it, so if you don't mind then lets keep retuning null here :-)

Reply:
@sberyozkin will be done :) i mixed up the apply with outside call where exceptions are handled.